source/components/triggering-changes-with-actions.md

```diff
@@ -0,0 +1,277 @@
+
+Let's take it step by step.
+
+#### Implementing the Action
+
+First, we'll define what we want to happen when the user double clicks
+the button. In this case, we'll find the user's account and delete it.
+
+Where do you put the code for an action? In Ember, each component can
+have a property called `actions`, where you put functions that can be
+triggered by child components.
+
+Let's look at the parent component's JavaScript file. In this example,
+imagine we have a parent component called `user-profile` that shows the
+user's profile to them.
+
+We'll implement an action on the parent component called
+`userDidDeleteAccount()` that, when called, gets a hypothetical `login`
+service and calls the service's `deleteUser()` method.
+
+```app/components/user-profile.js
+export default Ember.Component.extend({
+  login: Ember.inject.service(),
+
+  actions: {
+    userDidDeleteAccount() {
+      this.get('login').deleteUser();
+    }
+  }
+});
+```
+
+Now we've implemented our action, but we have not told Ember when we
+want this action to be triggered, which is the next step.
+
+#### Passing the Action to the Component
+
+What comes next is arguably the most important part of building a
+component, since it defines the public API: deciding what to call an
+action when it gets passed in.
+
+In this example, we want something to happen whenever the component is
+double clicked. So, when we give the action to the component, we're
+going to call it `onDoubleClick`:
+
+```app/components/user-profile.hbs
+{{double-click-button onDoubleClick=(action 'userDidDeleteAccount')}}
+```
+
+This snippet says "take the `userDidDeleteAccount` action from the
+parent and make it available on the child component as
+`onDoubleClick`."
+
+### Triggering the Action
+
+Next, how do we detect that a user has double clicked our component?
+Ember has built-in support for common browser events like this. In our
+component's JavaScript, we can implement a method called
+`doubleClick()`. Every time the user double clicks this component, Ember
+will call the `doubleClick()` method.
+
+```app/components/double-click-button.js
+export default Ember.Component.extend({
+  tagName: 'button',
+
+  // This method is called every time the user double clicks the
+  // component.
+  doubleClick() {
+  }
+});
+```
+
+Now that we've implemented the `doubleClick()`  method, we still need to
+trigger the `userDidDeleteAccount` action that we've been given.  Let's
+do that now.
+
+When you provide an action, it get turned into a method on the component
+you can call. Above we said:
+
+```hbs
+onDoubleClick=(action 'userDidDeleteAccount')`
+```
+
+That means that calling the component's `onDoubleClick` method will call
+the parent's `userDidDeleteAccount` action.
+
+```app/components/double-click-button.js
+doubleClick() {
+  // Call the component's `onDoubleClick()` method to trigger
+  // the parent's `userDidDeleteAccount` action.
+  this.onDoubleClick();
```

Comment:
`this.sendAction('onDoubleClick')` is consistent with the strings-passing world and not deprecated. If you wanted to directly address the action function, I think you would need to suggest `this.get('onDoubleClick')();`